src/main/java/core/di/factory/BeanFactory.java

```diff
@@ -33,19 +36,20 @@ public void initialize() {
         }
     }
 
-    private Object makeBean(Class<?> findClass) {
+    private Object makeBean(Class findClass) {
         findClass = BeanFactoryUtils.findConcreteClass(findClass, preInstanticateBeans);
 
-        if (!Optional.ofNullable(BeanFactoryUtils.getInjectedConstructor(findClass)).isPresent()) {
+        if (!existInjectConstructor(findClass)) {
             Object instance = BeanUtils.instantiateClass(findClass);
             this.beans.put(findClass, instance);
             return instance;
         }
 
         return injectBean(findClass);
+
     }
 
-    private Object injectBean(Class<?> injectClass) {
+    private Object injectBean(Class injectClass) {
```

Comment:
제네릭 타입을 직접적으로 명시해보는 건 어떨까요?
```suggestion
    private <T> T injectBean(Class<T> injectClass) {
```